zipline/algorithm.py

```diff
@@ -1106,7 +1109,12 @@ def schedule_function(self,
         # Check the type of the algorithm's schedule before pulling calendar
         # Note that the ExchangeTradingSchedule is currently the only
         # TradingSchedule class, so this is unlikely to be hit
-        cal = calendar or self.trading_calendar
+        if calendar is calendars.US_EQUITIES:
+            cal = get_calendar('NYSE')
+        elif calendar is calendars.US_FUTURES:
+            cal = get_calendar('us_futures')
+        else:
```

Comment:
Ah good catch. I'll add a check for `None`, and the `else` case should probably raise an exception.